onadata/apps/logger/models/version.py

```diff
@@ -0,0 +1,71 @@
+from django.db import models
+
+
+class VersionTreeException(Exception):
+    pass
+
+
+class VersionManager(models.Manager):
+    def get_root_path(self, vt):
+        """Get path from root to given vt"""
+        path = []
+        while vt is not None:
+            path.insert(0, vt)
```

Comment:
Insert is O(n). You want to literally do .append and return reversed list instead, because right now get_root_path is O(root_path_length^2)

Reply:
That's non-intuitive, I thought that inserting at the beggining of the list is O(1), like in other languages, glad you checked/knew that. Changed.